### PR TITLE
Send full update when a service is found to have no endpoints

### LINKED ISSUE #14336: Stale endpoints on scale to zero
Scale deployment to 5 replicas, see all 5 in `/clusters`

```
outbound|80||identity.envoylite.svc.cluster.local::10.60.22.51:8080::region::us-central1
outbound|80||identity.envoylite.svc.cluster.local::10.60.22.52:8080::region::us-central1
outbound|80||identity.envoylite.svc.cluster.local::10.60.25.75:8080::region::us-central1
outbound|80||identity.envoylite.svc.cluster.local::10.60.25.76:8080::region::us-central1
outbound|80||identity.envoylite.svc.cluster.local::10.60.25.77:8080::region::us-central1
```

Scale deployment to 0 replicas, endpoints still remain in `/clusters`

```
outbound|80||identity.envoylite.svc.cluster.local::10.60.22.51:8080::region::us-central1
outbound|80||identity.envoylite.svc.cluster.local::10.60.22.52:8080::region::us-central1
outbound|80||identity.envoylite.svc.cluster.local::10.60.25.75:8080::region::us-central1
outbound|80||identity.envoylite.svc.cluster.local::10.60.25.76:8080::region::us-central1
outbound|80||identity.envoylite.svc.cluster.local::10.60.25.77:8080::region::us-central1
```

Running 1.2 dailies from 5/22

Adding `s.ConfigUpdate(true)` to https://github.com/istio/istio/blob/master/pilot/pkg/proxy/envoy/v2/eds.go#L503 fixes this, but I am not sure it is the best option

Scaling back up triggers a push which resolves this

This is not very critical, because if you have 0 endpoints you aren't going to have any successful requests anyways

### PR DESCRIPTION
**Pull request by XanderStrike**
otherwise these endpoints will never be culled from sidecars

Please provide a description for what this PR is for.

resolves #14336

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/eds.go [hunk 1]
```diff
@@ -462,6 +463,8 @@ func (s *DiscoveryServer) edsUpdate(clusterID, serviceName string, namespace str
 			if svcShards == 0 {
 				delete(s.EndpointShardsByService[serviceName], namespace)
 			}
+			adsLog.Infof("Full push, service %s has no endpoints", serviceName)
+			s.ConfigUpdate(&model.PushRequest{Full: true})
```
Comment (hzxuzhonghu) [inline pilot/pkg/proxy/envoy/v2/eds.go]: Can you also set TargetNamespace?

Comment (howardjohn) [reply pilot/pkg/proxy/envoy/v2/eds.go]: Huh, I could have sworn this did have this set when I looked earlier... @XanderStrike did you remove it?

Comment (XanderStrike) [reply pilot/pkg/proxy/envoy/v2/eds.go]: I added it as a premature optimization, but unfortunately I discovered while manually testing that with namespace specified it doesn't actually send an update. Not sure why. If you like I can dig into that further I'm happy to do it.

**Comment by howardjohn** [comment]
maybe target namespaces looks at Services() and because the service is
deleted it's not in the list, so it is always skipped?

**Comment by XanderStrike** [comment]
Figured it out! I believe I was providing an _empty_ list of serviceNames, whereas the logic only sends full updates when the list is _not present_. It was iterating through an empty set, and therefore sending no updates.

The code is now scoped by namespace as requested.

**Comment by howardjohn** [comment]
/retest

will open an issue for these failures

**Pull request merged**